api/variables.go

```diff
@@ -380,6 +421,24 @@ type VariableMetadata struct {
 
 	// ModifyTime is the unix nano of the last modified time
 	ModifyTime int64 `hcl:"modify_time"`
+
+	// Lock holds the information about the variable lock if its being used.
+	Lock *VariableLock `hcl:",lock,optional" json:",omitempty"`
+}
+
+type VariableLock struct {
+	// ID is generated by Nomad to provide a unique caller ID which can be used
+	// for renewals and unlocking.
+	ID string
+
+	// TTL describes the time-to-live of the current lock holder.
+	// This is a string version of a time.Duration like "2m".
```

Comment:
might be worth adding the defaults values for TTL/LockDelay in these comments?  since for some users this may be their main interaction with the API.  and/or mention that they're optional.

Reply:
Because the api is a light client, I dont want to add much logic here, the defaults will be set during validation. (Im trying to keep it in line with everything else)